framework/tst/dslabs/framework/testing/visualization/VizClient.java

```diff
@@ -73,24 +74,26 @@
     private DataOutputStream out;
 
     private final SearchState state;
+    private final SearchSettings settings;
     private final StatePredicate invariant;
     private final String[] nodeNames;
     private final boolean trace;
 
     public VizClient(String host, int port, SearchState state,
-                     StatePredicate invariant, String[] nodeNames,
+                     SearchSettings settings, String[] nodeNames,
                      boolean trace) {
         this.host = host;
         this.port = port;
         this.state = state;
-        this.invariant = invariant;
+        this.settings = settings;
+        this.invariant = settings == null ? null : settings.whichInvariantViolated(state);
```

Comment:
If there is more than one invariant violation, the invariant which gets printed by the `BaseJUnitTest` could be different from the one returned here. But, we never actually use this field because Oddity never implemented anything for invariants. We're going to rip all of this out anyways, so this is fine for now.